eos-updater-avahi/eos-updater-avahi.c

```diff
@@ -23,12 +23,120 @@
 #include <glib.h>
 #include <libeos-update-server/config.h>
 #include <libeos-updater-util/avahi-service-file.h>
+#include <libeos-updater-util/extensions.h>
 #include <libeos-updater-util/ostree.h>
 #include <locale.h>
 #include <ostree.h>
 #include <stdlib.h>
 
 static gboolean
+get_refs (OstreeRepo     *repo,
+          gchar        ***out_refs,
+          GCancellable   *cancellable,
+          GError        **error)
+{
+  g_autoptr(GHashTable) refs = NULL;
+  GHashTableIter iter;
+  gchar *key;
+  gchar *value;
+  g_autoptr(GPtrArray) refs_array = NULL;
+
+  if (!ostree_repo_list_refs (repo,
+                              NULL, /* no prefix filtering */
+                              &refs,
+                              cancellable,
+                              error))
+    return FALSE;
+
+  if (g_hash_table_size (refs) == 0)
+    {
+      g_set_error_literal (error, G_IO_ERROR, G_IO_ERROR_FAILED,
+                           "no refs to advertise");
+      return FALSE;
+    }
+  refs_array = g_ptr_array_new_full (g_hash_table_size (refs) + 1, g_free);
+  g_hash_table_iter_init (&iter, refs);
+  while (g_hash_table_iter_next (&iter, (gpointer *)&key, (gpointer *)&value))
+    {
+      g_hash_table_iter_steal (&iter);
+      g_free (g_steal_pointer (&value));
+      g_ptr_array_add (refs_array, g_steal_pointer (&key));
+    }
+  g_ptr_array_add (refs_array, NULL);
+
+  g_assert (out_refs);
+  *out_refs = (gchar **)g_ptr_array_free (g_steal_pointer (&refs_array), FALSE);
+  return TRUE;
+}
+
+static gboolean
+get_summary_timestamp (OstreeRepo    *repo,
+                       GDateTime    **out_summary_timestamp,
+                       GCancellable  *cancellable,
+                       GError       **error)
+{
+  // TODO: Find the summary without using eos specific bits to ease
```

Comment:
Make this a `FIXME`, since I’ve been trying to follow the convention in eos-updater that `TODO` is something to fix before a patch is merged, and `FIXME` is something longer-term to fix eventually.